### DISCUSSION TITLE
Create A Next.js app - Installation failed

### DISCUSSION THREAD

**Discussion by vigneshvelu** [vercel/next.js#12884] [Help]
Hey everyone,

I am learning next.js. I am pretty sure I didn't describe the header properly. Sorry about that. 

I am following the tutorial in next.js website under 'Learn' section. ([Link](https://nextjs.org/learn/basics/create-nextjs-app/setup)) Unfortunately I am not able to run the first command. I am not create starter template. Below is the error I am getting in the terminal. My node version is the latest LTS version. Can someone please help me with this? Thank you:) I am running this in macos.

`
Desktop xperian$ npm init next-app nextjs-blog --example "https://github.com/zeit/next-learn-starter/tree/master/learn-starter"
npx: installed 1 in 1.363s
Creating a new Next.js app in /Users/xperian/Desktop/nextjs-blog.

Downloading files from repo https://github.com/zeit/next-learn-starter/tree/master/learn-starter. This might take a moment.

Installing packages. This might take a couple of minutes.

> fsevents@2.1.3 install /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/fork-ts-checker-webpack-plugin/node_modules/chokidar/node_modules/fsevents
> node-gyp rebuild

gyp: binding.gyp not found (cwd: /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/fork-ts-checker-webpack-plugin/node_modules/chokidar/node_modules/fsevents) while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/xperian/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:321:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.4.0
gyp ERR! command "/Users/xperian/node_modules/node/bin/node" "/Users/xperian/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/fork-ts-checker-webpack-plugin/node_modules/chokidar/node_modules/fsevents
gyp ERR! node -v v13.7.0
gyp ERR! node-gyp -v v3.6.0
gyp ERR! not ok
\
> fsevents@2.1.3 install /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/sass/node_modules/chokidar/node_modules/fsevents
> node-gyp rebuild

gyp: binding.gyp not found (cwd: /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/sass/node_modules/chokidar/node_modules/fsevents) while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/xperian/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:321:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.4.0
gyp ERR! command "/Users/xperian/node_modules/node/bin/node" "/Users/xperian/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/sass/node_modules/chokidar/node_modules/fsevents
gyp ERR! node -v v13.7.0
gyp ERR! node-gyp -v v3.6.0
gyp ERR! not ok

> fibers@5.0.0 install /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/fibers
> node build.js || nodejs build.js

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/xperian/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:321:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.4.0
gyp ERR! command "/Users/xperian/node_modules/node/bin/node" "/Users/xperian/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
gyp ERR! cwd /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/fibers
gyp ERR! node -v v13.7.0
gyp ERR! node-gyp -v v3.6.0
gyp ERR! not ok
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++ build-essential`
RHEL users please run: `yum install gcc-c++` and `yum groupinstall 'Development Tools'`
Alpine users please run: `sudo apk add python make g++`
sh: nodejs: command not found

> fsevents@1.2.13 install /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/fsevents
> node install.js

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/xperian/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:321:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.4.0
gyp ERR! command "/Users/xperian/node_modules/node/bin/node" "/Users/xperian/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/fsevents
gyp ERR! node -v v13.7.0
gyp ERR! node-gyp -v v3.6.0
gyp ERR! not ok
\
> node-sass@4.14.1 install /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/node-sass
> node scripts/install.js

Cached binary found at /Users/xperian/.npm/node-sass/4.14.1/darwin-x64-79_binding.node

> node-sass@4.14.1 postinstall /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/node-sass
> node scripts/build.js

Binary found at /Users/xperian/Desktop/nextjs-blog/node_modules/next/node_modules/node-sass/vendor/darwin-x64-79/binding.node
Testing binary
Binary is fine
npm ERR! Darwin 19.4.0
npm ERR! argv "/Users/xperian/node_modules/node/bin/node" "/Users/xperian/node_modules/.bin/npm" "install" "--loglevel" "error"
npm ERR! node v13.7.0
npm ERR! npm  v2.15.12
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn

npm ERR! fibers@5.0.0 install: `node build.js || nodejs build.js`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the fibers@5.0.0 install script 'node build.js || nodejs build.js'.
npm ERR! This is most likely a problem with the fibers package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build.js || nodejs build.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs fibers
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls fibers
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/xperian/Desktop/nextjs-blog/npm-debug.log

Aborting installation.
  npm install --loglevel error has failed.`

**Comment by sambhav-gore** [comment]
What is your node version ?

**Reply by vigneshvelu** [reply]
v12.16.3

**Comment by lfades** [comment]
Hi @vigneshvelu This is not related to Next.js but to `node-gyp` instead. You seem to be missing some required build tools for it, I'm not sure of what those are as there are many different solutions in google (And I don't have a macbook). I would recommend to follow the [node-gyp instructions for macOS](https://github.com/nodejs/node-gyp/blob/master/macOS_Catalina.md#the-acid-test), if that's not enough then google around and once you find a solution comment it here!
